signature-envelope-cose.md

```diff
@@ -0,0 +1,134 @@
+# COSE Sign1 Envelope Serialization
+
+In COSE ([rfc8152](https://datatracker.ietf.org/doc/html/rfc8152)), data is stored as either payload or headers (protected and unprotected).
+Notary v2 supports [COSE_Sign1_Tagged](https://datatracker.ietf.org/doc/html/rfc8152#section-4.2) as a signature envelope with some additional constraints on the header fields.
+
+A JWS based signature will be persisted with a blob `mediaType` of `"application/cose"`
+
+```jsonc
+{
+   "artifactType": "application/vnd.cncf.notary.v2.signature",
+    "blobs": [
+        {
+            "mediaType": "application/cose",
+            "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0",
+            "size": 32654
+        }
+    ],
+    "subject": {
+        "mediaType": "application/vnd.oci.image.manifest.v1+json",
+        "digest": "sha256:73c803930ea3ba1e54bc25c2bdc53edd0284c62ed651fe7b00369da519a3c333",
+        "size": 16724
+    },
+    "annotations": {
+        "io.cncf.notary.x509certs.fingerprint.sha256": "[\"B7A69A70992AE4F9FF103EBE04A2C3BA6C777E439253CE36562E6E98375068C3\" \"932EB6F5598435D4EF23F97B0B5ACB515FAE2B8D8FAC046AB813DDC419DD5E89\"]"
+    }
+}
+```
+
+Unless explicitly specified as OPTIONAL, all fields are required.
+
+**Payload**: COSE signs the payload as defined in the [Payload](#payload) section. Detached payloads are not supported.
+
+**ProtectedHeaders**: Notary v2 supports the following protected headers. Other header fields can be included but will be ignored.
+
+```
+{
+  / crit / 2: [
+    / cty / 3,
+    'signingtime',
+    'expiry'
+  ],
+  / cty / 3: 'application/vnd.cncf.oras.artifact.descriptor.v1+json',
+  'signingtime': 1234567891000,
+  'expiry': 1234567891011
+}
+```
+
+Note: The above example is represented using the [extended CBOR diagnostic notation](https://datatracker.ietf.org/doc/html/rfc8152#appendix-C).
+
+- **`crit`** (*array of integers or strings*): This REQUIRED property (label: `2`) lists the headers that implementation MUST understand and process.
+  The array MUST contain `3` (`cty`), and `signingtime`. If `expiry` is presented, the array MUST also contain `expiry`.
+- **`cty`** (*string*): The REQUIRED property content-type (label: `3`) is used to declare the media type of the secured content (the payload).
+- **`signingtime`** (*datetime*): The REQUIRED property identifies the time at which the signature was generated.
```

Comment:
Should be more specific on the format. `*datetime*` isn't defined anywhere.

Reply:
@letmaik, which RFC should we reference here?

Reply:
Its value is the number of seconds from `1970-01-01T00:00Z` in UTC time, commonly known as UNIX timestamp